### ISSUE TITLE
Text color in disabled text fields in Safari is wrong

### ISSUE THREAD

**Issue by MPor80** [microsoft/fluentui#20076]
### Environment Information

- **Package version(s)**: fluentui/react": 8.29.2
- **Browser and OS versions**: Safari version: 14.1.1 (16611.2.7.1.4)

### Please provide a reproduction of the bug in a codepen:
https://codepen.io/mayp80/pen/OJgqGjW

The text color in disabled text fields is different between Chrome and Safari with style:
field: {
        backgroundColor: "unset"
      }

Open the codepen in both chrome and safari and observe the difference:
Chrome:
![image](https://user-images.githubusercontent.com/91671706/135746297-997bbeaf-0b55-43b6-9e0a-b9677fa63456.png)

Safari:
![safari codepen](https://user-images.githubusercontent.com/91671706/135746331-91e29afb-e287-4b1e-8818-30e647db50d4.png)

#### Actual behavior:

The color in Safari is wrong and can barely be seen, even thought in devtools the right color is displayed.

#### Expected behavior:

The disabled text color in Safari should be correct.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Production blocking

Products/sites affected: Power Query for Mac

Thanks!

**Comment by gouttierre** [comment]
@MPor80 - Thanks for filing this issue with us. To set expectation our team will have a look at the issue but only once capacity allows.

@bsunderhus -  Would you be able to confirm if this is a regression, or if this behavior is an issue on Safari?
cc @smhigley

**Comment by odhanson** [comment]
@gouttierre this is an accessibility bug which is blocking us from opening an Office feature to Production, also we have an SLA for this. So it would help to prioritize it if its an easy fix.

**Comment by GeoffCoxMSFT** [comment]
@odhanson @MPor80 - I can reproduce this on a Mac/Safari using your CodePen.  

The problem occurs because of how the backgroundColor is set to 'unset' regardless of props.  The styles attributes wins over the default styling so this sets the backgroundColor to 'unset' for all states disabled, focused, required, etc.  Because the disabled colors from the theme are meant to be used together, this causes insufficient contrast between the background and text colors.

To set the background to unset and meet accessibility, you will need to only apply this in the default case.  You could choose to have the background unset in the focused case too.  

```
const textFieldStyles = (props: ITextFieldStyleProps): Partial<ITextFieldStyles> => {
  const { disabled, focused } = props;
  return {
    field: [
      !disabled && !focused && {
        backgroundColor: 'unset',
      },
    ],
  };
};
```

I would recommend you do the same thing to color as you do to backgroundColor so they are paired from either the browser default settings or their parents via cascading styles). A browser in light/dark mode may not have a default background that has sufficient contrast with the color for that theme.
